.gitlab/butte-jobs.yml

```diff
@@ -5,92 +5,67 @@
 # SPDX-License-Identifier: (MIT)
 ##############################################################################
 
-####
-# Shared configuration of jobs for butte
-.on_butte_advanced:
-  variables:
-  tags:
-    - shell
-    - butte
-  rules:
-    - if: '$ON_BUTTE != "ON"' #run except if ...
-      when: never
-    - when: on_success
-  allow_failure: true
-
-.build_and_test_on_butte:
-  stage: b_build_and_test
-  extends: [.build_blueos_3_ppc64le_ib_script, .on_butte_advanced]
-  needs: []
-
-# Note: .build_and_test_on_butte_advanced_advanced inherits from
-# .build_and_test_on_butte_advanced and .advanced_pileline.
-# In particular, the rules section will be merged. Careful when changing rules.
-.build_and_test_on_butte_advanced:
-  extends: [.build_and_test_on_butte, .advanced_pipeline]
-  allow_failure: true
-
 ####
 # All butte build jobs
 
 clang_default (build and test on butte):
   variables:
-    HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-clang@default.cmake"
+    SPEC: "%clang@default"
   extends: .build_and_test_on_butte_advanced
 
 gcc_default (build and test on butte):
   variables:
-    HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-gcc@default.cmake"
+    SPEC: "%gcc@default"
   extends: .build_and_test_on_butte_advanced
 
 xl_default (build and test on butte):
   variables:
-    HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-xl@default.cmake"
+    SPEC: "%xl@default"
   extends: .build_and_test_on_butte_advanced
 
 pgi_default (build and test on butte):
   variables:
-    HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-pgi@default.cmake"
+    SPEC: "%pgi@default"
   extends: .build_and_test_on_butte_advanced
   allow_failure: true
 
-clang_3_9_1 (build and test on butte):
+gcc_8_3_1 (build and test on butte):
   variables:
-    HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-clang@3.9.1.cmake"
+    SPEC: "%gcc@8.3.1"
   extends: .build_and_test_on_butte_advanced
 
-clang_4_0_0 (build and test on butte):
+clang_3_9_1 (build and test on butte):
   variables:
-    HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-clang@4.0.0.cmake"
+    SPEC: "%clang@3.9.1"
   extends: .build_and_test_on_butte_advanced
 
-clang_9_0_0 (build and test on butte):
+clang_4_0_0 (build and test on butte):
   variables:
-    HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-clang@9.0.0.cmake"
+    SPEC: "%clang@4.0.0"
   extends: .build_and_test_on_butte_advanced
 
-gcc_8_3_1 (build and test on butte):
+clang_9_0_0_libcpp (build and test on butte):
   variables:
-    HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-gcc@8.3.1.cmake"
+    SPEC: "%clang@9.0.0"
```

Comment:
Hey @adrienbernede - this is the only thing I'm confused about, which is the spec for clang_9 with libc++ `%clang%9.0.0`? Shouldn't the libc++ be a variant?

Reply:
Good catch!
That’s something I want to solve in the future: there is no guarantee that the name of the job describe the spec.

Reply:
Okay, I see. So in this case, the build is clang9, but the job is just named wrong? (it's not a problem, just wondering)

Reply:
Fixed.
Note: this wasn’t in the CI list of specs before, I may have introduced it to show-case variant use, in replacement of clang@9.0.0 alone. If you want to keep clang@9.0.0 too, just tell me so.

Reply:
I added the variant (changed the spec) instead of changing the job name :).